src/data/repositories/InstanceDhisRepository.ts

```diff
@@ -22,9 +23,12 @@ export class InstanceDhisRepository implements InstanceRepository {
         const blob = new Blob([data], { type: mime });
         const resized = mime.startsWith("image") ? await resizeFile(blob) : blob;
 
+        const fileData = await arrayBufferToString(data);
+        const fileId = getUid(fileData);
```

Comment:
[minir] Irrelevant for the current usage, but we only need `fileData/fileId` (costy operation) if `options.id` is not present: `id: options.id ?? getFileId(data)`.

Reply:
Noted, will do on next release